crates/common/src/commit/client.rs

```diff
@@ -83,17 +91,58 @@ impl SignerClient {
             });
         }
 
-        let signature: BlsSignature = serde_json::from_slice(&response_bytes)?;
+        let signature: Vec<u8> = serde_json::from_slice(&response_bytes)?;
```

Comment:
This is only for quick type unification inside the client SDK, so I'd say it's whatever.

However, the REST API does currently include a lot of not-so-nicely serialised stuff whilst we're experimenting with different approaches, so if that's your concern, it will have to be polished up at the end, sure. Added as a PR task.

Reply:
Intermediate `Vec<u8>` deserialization removed in [b8f228b](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/b8f228b630f1865439959b166b5a87a5a23bc0b9). 

Rest API hex encoding of request/response structs fixed in [dd864ec](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/dd864ec2d2fe7e2e3a53003c511431a51388bb1e)